io/EptReader.cpp

```diff
@@ -381,64 +418,74 @@ void EptReader::readLaszip(PointView& dst, const Key& key) const
     std::lock_guard<std::mutex> lock(m_mutex);
     reader.prepare(table);
 
-    // While the laszip data type will contain a scale/offset in the Entwine
-    // metadata, the LasReader will handle the transformation.
     for (auto& src : reader.execute(table))
     {
         PointRef pr(*src);
         for (uint64_t i(0); i < src->size(); ++i)
         {
             pr.setPointId(i);
-            process(dst, pr, false);
+            process(dst, pr);
         }
     }
 }
 
 void EptReader::readBinary(PointView& dst, const Key& key) const
 {
-    auto data(m_ep->getBinary(key.toString() + ".bin"));
+    auto data(m_ep->getBinary("ept-data/" + key.toString() + ".bin"));
     ShallowPointTable table(*m_remoteLayout, data.data(), data.size());
     PointRef pr(table);
 
     std::lock_guard<std::mutex> lock(m_mutex);
 
-    // Binary data scale/offset needs to be handled here.
     for (uint64_t i(0); i < table.numPoints(); ++i)
     {
         pr.setPointId(i);
-        process(dst, pr, true);
+        process(dst, pr);
     }
 }
 
-void EptReader::process(PointView& dst, PointRef& pr, bool unscale) const
+void EptReader::process(PointView& dst, PointRef& pr) const
 {
-    const auto pointId(dst.size());
-    double x = pr.getFieldAs<double>(Dimension::Id::X);
-    double y = pr.getFieldAs<double>(Dimension::Id::Y);
-    double z = pr.getFieldAs<double>(Dimension::Id::Z);
-    const uint64_t o = pr.getFieldAs<uint64_t>(Dimension::Id::OriginId);
-
-    if (unscale)
+    auto dimIndex = [this](Dimension::Id id)->uint64_t
     {
-        x = x * m_info->scale()[0] + m_info->offset()[0];
-        y = y * m_info->scale()[1] + m_info->offset()[1];
-        z = z * m_info->scale()[2] + m_info->offset()[2];
-    }
+        for (uint64_t i(0); i < m_remoteLayout->dims().size(); ++i)
+        {
+            if (m_remoteLayout->dims()[i] == id)
+            {
+                return i;
+            }
+        }
+        throw pdal_error("Invalid dimIndex lookup");
+    };
+
+    using D = Dimension::Id;
+    const std::array<XForm, 3> xforms { {
+        m_dimTypes[dimIndex(D::X)].m_xform,
+        m_dimTypes[dimIndex(D::Y)].m_xform,
+        m_dimTypes[dimIndex(D::Z)].m_xform,
+    } };
+
+    const point_count_t pointId(dst.size());
+
+    const double x = pr.getFieldAs<double>(Dimension::Id::X) *
+        xforms[0].m_scale.m_val + xforms[0].m_offset.m_val;
+    const double y = pr.getFieldAs<double>(Dimension::Id::Y) *
+        xforms[1].m_scale.m_val + xforms[1].m_offset.m_val;
+    const double z = pr.getFieldAs<double>(Dimension::Id::Z) *
+        xforms[2].m_scale.m_val + xforms[2].m_offset.m_val;
+
+    const uint64_t o = pr.getFieldAs<uint64_t>(Dimension::Id::OriginId);
 
     const bool selected = !m_queryOriginId || o == *m_queryOriginId;
```

Comment:
This is an offset into the sources, right?  You could set it to -1 rather than mess with the pointer thing, right?

Reply:
Yes.